test/parallel/test-stream-finished.js

```diff
@@ -396,3 +403,67 @@ testClosed((opts) => new Writable({ write() {}, ...opts }));
   r.destroyed = true;
   r.push(null);
 }
+
+{
+  // Regression https://github.com/nodejs/node/issues/33130
+
+  const response = new PassThrough();
+  setTimeout(() => response.write('chunk 1'), 500);
+  setTimeout(() => response.write('chunk 2'), 1000);
+  setTimeout(() => response.write('chunk 3'), 1500);
+  setTimeout(() => response.end(), 2000);
```

Comment:
Can we do this without timeouts? They generally shouldn’t be necessary for streams tests, I think